openpathsampling/analysis/shooting_point_analysis.py

```diff
@@ -138,8 +143,25 @@ def analyze_single_step(self, step):
                  for state in self.states}
             )
             total_count = sum(total.values())
-            # TODO: clarify assertion (at least one endpoint in state)
-            assert total_count == 1 or total_count == 2
+            if total_count == 0:
+                err = ("Step "+str(step.mccycle)+" has a trajectory without "
+                       "endpoints in any of the states.")
+                solvable = (not step.change.canonical.accepted and
+                            not self.skip_rejected)
+                if solvable:
+                    err += ("\nTo skip analysis of rejected steps set "
+                            "'skip_rejected=True'")
+                raise ValueError(err)
+            if total_count > len(test_points):
+                # TODO: is this actually wrong / an error?
+                err = ("The " + str(len(test_points)) +
+                       " end points of the trail trajectory from step " +
+                       str(step.mccycle) +
+                       " found " + str(total_count) + " stable states."
+                       "\n Are you sure your states don't overlap?"
+                       )
+                raise ValueError(err)
```

Comment:
@dwhswenson I implemented this to be consistent with the (idea of) original assertion (check if you find more states than tested points). However, I am not convinced this actually wrong/ something we should raise on.

Is there any analysis that breaks if you snaps do not uniquely map to a single state?